osu.Game/Online/HubClientConnector.cs

```diff
@@ -144,13 +147,22 @@ private HubConnection buildConnection(CancellationToken cancellationToken)
                     options.Headers.Add("OsuVersionHash", versionHash);
                 });
 
-            if (RuntimeInfo.SupportsJIT)
+            if (RuntimeInfo.SupportsJIT && preferMessagePack)
                 builder.AddMessagePackProtocol();
             else
             {
                 // eventually we will precompile resolvers for messagepack, but this isn't working currently
                 // see https://github.com/neuecc/MessagePack-CSharp/issues/780#issuecomment-768794308.
-                builder.AddNewtonsoftJsonProtocol(options => { options.PayloadSerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore; });
+                builder.AddNewtonsoftJsonProtocol(options =>
+                {
+                    options.PayloadSerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
+                    options.PayloadSerializerSettings = new JsonSerializerSettings
```

Comment:
Looks as if the value set for `ReferenceLoopHandling` will be immediately discarded this way...?